writer.go

```diff
@@ -177,81 +167,32 @@ func (re *WriterResponseEmitter) Emit(v interface{}) error {
 	return re.enc.Encode(v)
 }
 
-type Any struct {
-	types map[reflect.Type]bool
-	order []reflect.Type
+type MaybeError struct {
+	Value interface{} // needs to be a pointer
+	Error cmdkit.Error
 
-	v interface{}
+	isError bool
 }
 
-func (a *Any) UnmarshalJSON(data []byte) error {
-	var (
-		iv  interface{}
-		err error
-	)
-
-	for _, t := range a.order {
-		v := reflect.New(t).Elem().Addr()
-
-		isNil := func(v reflect.Value) (yup, ok bool) {
-			ok = true
-			defer func() {
-				r := recover()
-				if r != nil {
-					ok = false
-				}
-			}()
-			yup = v.IsNil()
-			return
-		}
-
-		isZero := func(v reflect.Value, t reflect.Type) (yup, ok bool) {
-			ok = true
-			defer func() {
-				r := recover()
-				if r != nil {
-					ok = false
-				}
-			}()
-			yup = v.Elem().Interface() == reflect.Zero(t).Interface()
-			return
-		}
-
-		err = json.Unmarshal(data, v.Interface())
-
-		vIsNil, isNilOk := isNil(v)
-		vIsZero, isZeroOk := isZero(v, t)
-
-		nilish := (isNilOk && vIsNil) || (isZeroOk && vIsZero)
```

Comment:
It looks like we do actually need this logic.

Reply:
I don't get it. Why?